### PR TITLE
docs: update plugin-develop.md

### PR DESCRIPTION
**Pull request by shenzhuangzhi1**
### What this PR does / why we need it:

Missing content in Chinese document.
### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by shenzhuangzhi1** [comment]
what happened？

File: docs/zh/latest/plugin-develop.md [hunk 1]
```diff
@@ -37,6 +38,50 @@ title: 插件开发
 - [编写测试用例](#编写测试用例)
   - [附上 test-nginx 执行流程](#附上-test-nginx-执行流程)
 
+此文档是关于lua语言的插件开发,其他语言请看：[external plugin](./external-plugin.md)。
+
+## 插件放置路径
+
+wAPISIX提供了两种方式来添加新的功能。
```
Comment (tokers) [inline docs/zh/latest/plugin-develop.md:45]:
```suggestion
APISIX提供了两种方式来添加新的功能。
```

File: docs/zh/latest/plugin-develop.md [hunk 2]
```diff
@@ -37,6 +38,50 @@ title: 插件开发
...
+
+```yaml
+apisix:
+    ...
+    extra_lua_path: "/path/to/example/?.lua"
+    lua_module_hook: "my_hook"
+```
+
+ 当APISIX启动的时候，`example/my_hook.lua` 就会被加载，这时你可以使用这个钩子在APISIX中来全局替换掉一个方法。
+这个例子： [my_hook.lua](https://github.com/apache/apisix/blob/master/example/my_hook.lua) 可以在项目的`example`路径下被找到。
```
Comment (tokers) [inline docs/zh/latest/plugin-develop.md:84]:
```suggestion
这个例子： [my_hook.lua](https://github.com/apache/apisix/blob/master/example/my_hook.lua) 可以在项目的`example`路径下被找到。

```

File: docs/zh/latest/plugin-develop.md (hunk 1 above)
Comment (tzssangglass) [reply docs/zh/latest/plugin-develop.md:45]: APISIX 提供了两种方式来添加新的功能。

File: docs/zh/latest/plugin-develop.md [hunk 3]
```diff
@@ -37,6 +38,51 @@ title: 插件开发
 - [编写测试用例](#编写测试用例)
   - [附上 test-nginx 执行流程](#附上-test-nginx-执行流程)
 
+此文档是关于lua语言的插件开发,其他语言请看：[external plugin](./external-plugin.md)。
```
Comment (tzssangglass) [inline docs/zh/latest/plugin-develop.md:41]:
```suggestion
此文档是关于 lua 语言的插件开发，其他语言请看：[external plugin](./external-plugin.md)。
```

**Review by shenzhuangzhi1 — COMMENTED** [review]
update plugin-develop.md

**Comment by leslie-tsang** [comment]
Hello there, I'm afraid the syntax in the document needs to be optimized first, as the current version is a little strange.

Maybe you can refer to [copywriting-guidelines](https://github.com/sparanoid/chinese-copywriting-guidelines) to fix them.

File: docs/zh/latest/plugin-develop.md [hunk 4]
```diff
@@ -37,6 +38,53 @@ title: 插件开发
 - [编写测试用例](#编写测试用例)
   - [附上 test-nginx 执行流程](#附上-test-nginx-执行流程)
 
+此文档是关于 lua 语言的插件开发，其他语言请看：[external plugin](./external-plugin.md)。
+
+## 插件放置路径
+
+APISIX提供了两种方式来添加新的功能。
```
Comment (zaunist) [inline docs/zh/latest/plugin-develop.md:45]:
```suggestion
Apache APISIX 提供了两种方式来添加新的功能。
```

File: docs/zh/latest/plugin-develop.md [hunk 5]
```diff
@@ -37,6 +38,53 @@ title: 插件开发
...
+
+1. 修改APISIX的源代码并重新发布 (不推荐)。
```
Comment (zaunist) [inline docs/zh/latest/plugin-develop.md:47]:
```suggestion
1. 修改 Apache APISIX 的源代码并重新发布 (不推荐)。
```
There should be a space between Chinese and English, you could ref https://github.com/sparanoid/chinese-copywriting-guidelines .

File: docs/zh/latest/plugin-develop.md [hunk 6]
```diff
@@ -37,6 +38,53 @@ title: 插件开发
...
+
+```yaml
+apisix:
+    ...
+    extra_lua_path: "/path/to/example/?.lua"
+    lua_module_hook: "my_hook"
+```
+
+ 当APISIX启动的时候，`example/my_hook.lua` 就会被加载，这时你可以使用这个钩子在APISIX中来全局替换掉一个方法。
+这个例子： [my_hook.lua](https://github.com/apache/apisix/blob/master/example/my_hook.lua) 可以在项目的`example`路径下被找到。
```
Comment (tzssangglass) [inline docs/zh/latest/plugin-develop.md:84]:
```suggestion
这个例子： [my_hook.lua](https://github.com/apache/apisix/blob/master/example/my_hook.lua) 可以在项目的 `example` 路径下被找到。
```

**Review by tzssangglass — DISMISSED** [review]

**Review by zaunist — DISMISSED** [review]

File: docs/zh/latest/plugin-develop.md [hunk 7]
```diff
@@ -37,6 +38,51 @@ title: 插件开发
...
+
+```yaml
+apisix:
+    ...
+    extra_lua_path: "/path/to/example/?.lua"
+```
+
+现在使用 `require "apisix.plugins.3rd-party"` 会加载你自己的插件， 比如 `require "apisix.plugins.jwt-auth"`会加载 `jwt-auth` 插件.
+
+可能你会想覆盖一个文件中的函数，你可以在`conf/config.yaml`文件中配置`lua_module_hook` 来使你的hook生效。
```
Comment (leslie-tsang) [inline docs/zh/latest/plugin-develop.md:72]:
```suggestion
可能你会想覆盖一个文件中的函数，你可以在 `conf/config.yaml` 文件中配置 `lua_module_hook` 来使你的 hook 生效。
```

File: docs/zh/latest/plugin-develop.md [hunk 8]
```diff
@@ -37,6 +38,51 @@ title: 插件开发
...
+你的配置可以像下面这样:
+
+```yaml
+apisix:
+    ...
+    extra_lua_path: "/path/to/example/?.lua"
+    lua_module_hook: "my_hook"
+```
+
+ 当APISIX启动的时候，`example/my_hook.lua` 就会被加载，这时你可以使用这个钩子在APISIX中来全局替换掉一个方法。
```
Comment (leslie-tsang) [inline docs/zh/latest/plugin-develop.md:83]:
```suggestion
 当 APISIX 启动的时候，`example/my_hook.lua` 就会被加载，这时你可以使用这个钩子在 APISIX 中来全局替换掉一个方法。
```

File: docs/zh/latest/plugin-develop.md [hunk 9]
```diff
@@ -37,6 +38,51 @@ title: 插件开发
...
+这个例子： [my_hook.lua](https://github.com/apache/apisix/blob/master/example/my_hook.lua) 可以在项目的 `example` 路径下被找到。
```
Comment (leslie-tsang) [inline docs/zh/latest/plugin-develop.md:84]:
```suggestion
这个例子：[my_hook.lua](https://github.com/apache/apisix/blob/master/example/my_hook.lua) 可以在项目的 `example` 路径下被找到。
```

File: docs/zh/latest/plugin-develop.md [hunk 10]
```diff
@@ -37,6 +38,51 @@ title: 插件开发
...
   - [附上 test-nginx 执行流程](#附上-test-nginx-执行流程)
 
+此文档是关于 lua 语言的插件开发，其他语言请看：[external plugin](./external-plugin.md)。
+
+## 插件放置路径
+
+Apache APISIX 提供了两种方式来添加新的功能。
+
+1. 修改 Apache APISIX 的源代码并重新发布 (不推荐)。
+1. 配置  `extra_lua_path` 和 `extra_lua_cpath` 在 `conf/config.yaml`以加载你自己的代码文件。 你应该给自己的代码文件起一个不包含在原来库中的名字，而不是使用相同名称的代码文件，但是如果有需要，你可以使用这种方式覆盖内置的代码文件。
```
Comment (leslie-tsang) [inline docs/zh/latest/plugin-develop.md:48]:
```suggestion
2. 配置 `extra_lua_path` 和 `extra_lua_cpath` 在 `conf/config.yaml` 以加载你自己的代码文件。 你应该给自己的代码文件起一个不包含在原来库中的名字，而不是使用相同名称的代码文件，但是如果有需要，你可以使用这种方式覆盖内置的代码文件。
```

File: docs/zh/latest/plugin-develop.md [hunk 11]
```diff
@@ -37,6 +38,51 @@ title: 插件开发
...
+├── example
+│   └── apisix
+│       ├── plugins
+│       │   └── 3rd-party.lua
+│       └── stream
+│           └── plugins
+│               └── 3rd-party.lua
+```
+
+接着，在`conf/config.yaml`文件中添加如下的配置:
```
Comment (leslie-tsang) [inline docs/zh/latest/plugin-develop.md:62]:
```suggestion
接着，在 `conf/config.yaml` 文件中添加如下的配置:
```

**Review by leslie-tsang — CHANGES_REQUESTED** [review]

File: docs/zh/latest/plugin-develop.md (hunk 10 above)
Comment (leslie-tsang) [reply docs/zh/latest/plugin-develop.md:48]: > 你应该给自己的代码文件起一个不包含在原来库中的名字

It looks strange to me, can you try to optimize the translation?

File: docs/zh/latest/plugin-develop.md (hunk 8 above)
Comment (zaunist) [reply docs/zh/latest/plugin-develop.md:83]:
```suggestion
 当 Apache APISIX 启动的时候，`example/my_hook.lua` 就会被加载，这时你可以使用这个钩子在 Apache APISIX 中来全局替换掉一个方法。
```

**Comment by tzssangglass** [comment]
Just some formatting issues

**Comment by zaunist** [comment]
Could you use `Apache APISIX` instead of `APISIX` ?

**Comment by spacewander** [comment]
> 

Both APISIX & Apache APISIX are OK.

**Comment by spacewander** [comment]
We use APISIX directly in many places of the doc. And the original doc doesn't use Apache APISIX.

**Comment by zaunist** [comment]
Got it.

**Review by tzssangglass — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**